### PR TITLE
Ensures when offline map loads that it is at least partially visible.

### PR DESCRIPTION
**Pull request by esreli**
This PR addresses this enhancement request:
https://github.com/Esri/data-collection-ios/issues/152

### CODE REVIEW

File: data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Map.swift [hunk 1]
```diff
@@ -29,40 +29,49 @@ extension MapViewController {
...
+            self.delegate?.mapViewController(self, didUpdateTitle: map.item?.title ?? "Map")
             
             guard let operationalLayers = map.operationalLayers as? [AGSFeatureLayer] else {
-                self?.delegate?.mapViewController(self!, shouldAllowNewFeature: false)
+                self.delegate?.mapViewController(self, shouldAllowNewFeature: false)
                 return
             }
             
-            AGSLoadObjects(operationalLayers, { [weak self] (_) in
+            AGSLoadObjects(operationalLayers, { (_) in
```
Comment (philium) [inline data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Map.swift]: This closure should still weakly capture `self`.

**Comment by esreli** [comment]
@philium would you have another look?

File: data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Map.swift (hunk 1 above)
Comment (esreli) [reply data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Map.swift]: I wasn't under the impression this was needed. What am I missing? The retain count remains the same without capturing self weakly.

Comment (philium) [reply data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Map.swift]: The issue is that `completion` is an escaping closure. The default is to strongly capture `self`. But by doing so, the (potentially long running) asynchronous operation prevents `self` from being deallocated in the event that `self` is no longer needed. I suppose that may not make a difference if the map view controller is never deallocated during the lifetime of the app, but it is good practice to weakly capture `self` when performing asynchronous operations.

**Pull request merged**